client/src/connection_cache.rs

```diff
@@ -145,10 +145,10 @@ impl Default for ConnectionCache {
 }
 
 macro_rules! dispatch {
-    ($(#[$meta:meta])* $vis:vis fn $name:ident$(<$($t:ident: $cons:ident),*>)?(&self $(, $arg:ident: $ty:ty)*) $(-> $out:ty)?) => {
+    ($(#[$meta:meta])* $vis:vis fn $name:ident$(<$($t:ident: $cons:ident + ?Sized),*>)?(&self $(, $arg:ident: $ty:ty)*) $(-> $out:ty)?) => {
         #[inline]
         $(#[$meta])*
-        $vis fn $name$(<$($t: $cons),*>)?(&self $(, $arg:$ty)*) $(-> $out)? {
+        $vis fn $name$(<$($t: $cons + ?Sized),*>)?(&self $(, $arg:$ty)*) $(-> $out)? {
```

Comment:
I can't comment further down, but even though it isn't used, can you add the same `+ ?Sized` to the `&mut self` version too?